lib/rules/custom-property-pattern/index.js

```diff
@@ -40,30 +40,32 @@ const rule = (primary) => {
 		 */
 		function check(property) {
 			return (
-				!isStandardSyntaxProperty(property) ||
 				!isCustomProperty(property) ||
+				!isStandardSyntaxProperty(property) ||
 				regexpPattern.test(property.slice(2))
 			);
 		}
 
 		root.walkDecls((decl) => {
 			const { prop, value } = decl;
 
-			const parsedValue = valueParser(value);
+			if (/var\(/i.test(value)) {
```

Comment:
[suggest] Caching a regex seems to improve performance a bit. Any thoughts?

For example:
https://github.com/stylelint/stylelint/blob/518367a28c8e742516bc8882b6cdf0a843c15ff2/lib/rules/function-calc-no-unspaced-operator/index.js#L28